src/main/java/com/daisyflemming/restful/spring/boot/UIDService.java

```diff
@@ -0,0 +1,77 @@
+package com.daisyflemming.restful.spring.boot;
+
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+@RestController
+public class UIDService {
+
+    private char[] alphanumerics = generateAlphanumericCharters();
+    private int[] bases = {21, 36, 36, 36, 36, 36};
+    private int xMax = Arrays.stream(bases).reduce(1, (a, b) -> a * b) - 1;
+    private int maxLength = 6;
+    private int minLength = 4;
+    /**
+     * nextId will be cached. Ideally this should be stored in a database or a file
+     * so when spring reboot, it can be restored.
+     */
+    private int nextId = 0;
+
+    /**
+     * This method will take a query parameter length= minLength ... maxLength.
+     * If no parameter is given, maxLength will be used
+     *
+     * @param length is the length of character required for the id. If none is given, it is default to the max. length
+     * @return UID the contains both the result batch id and the input value
+     */
+    @RequestMapping(value = {"/getNextId", "/getNextId/{length}"}, method = RequestMethod.GET)
```

Comment:
It would also be good to add a method that has a request mapping for something like /id/{number} so that someone could call the service with localhost:8080/id/100 and get what the ID will be for 100.